Compiler/src/profiling/tracy.jl

```diff
@@ -0,0 +1,67 @@
+#=
+# TODO: Enable submodule, seems to hit some bug right now with === not defined?
+module Tracy
+
+import ..@noinline, ..Cint, ..Vector, ..push!, ..unsafe_convert, ..esc,
+       ..pointer_from_objref, ..String, ..Ptr, ..UInt8, ..Cvoid,
+       ..Expr, ..LineNumberNode, ..Symbol, ..UInt32
+=#
+
+_strpointer(s::String) = ccall(:jl_string_ptr, Ptr{UInt8}, (Any,), s)
+
+mutable struct TracySrcLoc
+    zone_name::Ptr{UInt8}
+    function_name::Ptr{UInt8}
+    file::Ptr{UInt8}
+    line::UInt32
+    color::UInt32
+    # Roots
+    zone_name_str::String
+    function_name_sym::Symbol
+    file_sym::Symbol
+end
+TracySrcLoc(zone_name::String, function_name::Symbol, file::Symbol, line::UInt32, color::UInt32) =
+    TracySrcLoc(_strpointer(zone_name),
+                unsafe_convert(Ptr{UInt8}, function_name),
+                unsafe_convert(Ptr{UInt8}, file),
+                line, color, zone_name, function_name, file)
```

Comment:
```suggestion
    TracySrcLoc(C_NULL, C_NULL, C_NULL, line, color, zone_name, function_name, file)
```

Optional, but simpler since this will be re-initialized anyway